starlette/datastructures.py

```diff
@@ -323,24 +330,39 @@ def __repr__(self) -> str:
         return f"{class_name}({items!r})"
 
 
-class MultiDict(ImmutableMultiDict[typing.Any, typing.Any]):
-    def __setitem__(self, key: typing.Any, value: typing.Any) -> None:
+class MultiDict(
+    ImmutableMultiDict[_KeyType, _ValueType],
+    typing.MutableMapping[_KeyType, _ValueType],
+):
+    def __setitem__(self, key: _KeyType, value: _ValueType) -> None:
         self.setlist(key, [value])
 
-    def __delitem__(self, key: typing.Any) -> None:
+    def __delitem__(self, key: _KeyType) -> None:
         self._list = [(k, v) for k, v in self._list if k != key]
         del self._dict[key]
 
-    def pop(self, key: typing.Any, default: typing.Any = None) -> typing.Any:
+    @typing.overload
+    def pop(self, __key: _KeyType) -> _ValueType:
+        ...  # pragma: no cover
+
+    @typing.overload
+    def pop(
+        self,
+        __key: _KeyType,
+        __default: typing.Union[_ValueType, _DefaultType],
```

Comment:
Why indicate that default is positional with `__`?

Reply:
It's what typeshed does for dict: https://github.com/python/typeshed/blob/master/stdlib/builtins.pyi#L1059

Reply:
Notice that the implementation still uses `default` so this won't break anything at runtime, I only changed the tests because the type checker will yell at you